ext/symengine/ruby_matrix.c

```diff
@@ -0,0 +1,123 @@
+#include "ruby_matrix.h"
+
+void cmatrix_dense_free(void *ptr)
+{
+    CDenseMatrix *mat_ptr = ptr;
+    dense_matrix_free(mat_ptr);
+}
+
+VALUE cmatrix_dense_alloc(VALUE klass)
+{
+    CDenseMatrix *mat_ptr = dense_matrix_new();
+    return Data_Wrap_Struct(klass, NULL, cmatrix_dense_free, mat_ptr);
```

Comment:
I was writing it by taking the example of cbasic_alloc. So assumed, the klass is something which is called when allocating the space bound defined in `rb_define_alloc_func(c_dense_matrix, cmatrix_dense_alloc);`

Shoud I remove it?